dfastmi/io/ConfigFileOperations.py

```diff
@@ -47,20 +55,211 @@ def write_config(filename: str, config: configparser.ConfigParser) -> None:
+        """
+        for key in ("RiverKM", "FigureDir", "OutputDir"):
+            if key in config["General"]:
+                ConfigFileOperations._update_to_relative_path(rootdir, config, "General", key)
+        for qstr in config.keys():
+            if "Reference" in config[qstr]:
+                ConfigFileOperations._update_to_relative_path(rootdir, config, qstr, "Reference")
+            if "WithMeasure" in config[qstr]:
+                ConfigFileOperations._update_to_relative_path(rootdir, config, qstr, "WithMeasure")
+        return config
+    
+    @staticmethod
+    def _update_to_relative_path(rootdir : str, config : configparser.ConfigParser, section : str, key : str):
+        """
+        Convert a configuration object to contain absolute paths (for editing).
+        Arguments
+        ---------
+        rootdir : str
+            The path to be used as base for the absolute paths.
+        config : configparser.ConfigParser
+            Configuration for the D-FAST Morphological Impact analysis with absolute paths 
+            to be converted to relative paths.
+        section : str
+            Where in the configuration we need to update the absolute path
+        key : str
+            Where in the configuration we need to update the absolute path
+
+        Returns
+        -------
+        aconfig : configparser.ConfigParser
+            Configuration for the D-FAST Morphological Impact analysis with only relative paths.
+        """
+        absolute_path = config.get(section, key, fallback="")
+        absolute_path_converted_to_relative_path = FileUtils.relative_path(rootdir, absolute_path)
+        config.set(section, key, absolute_path_converted_to_relative_path)
+
+    @staticmethod
+    def load_configuration_file(filename: str) -> configparser.ConfigParser:
+        """
+        Open a configuration file and return a configuration object with absolute paths.
+
+        Arguments
+        ---------
+        filename : str
+            The name of the file: all relative paths in the configuration will be assumed relative to this.
+
+        Raises
+        ------
+        Exception
+            If the configuration file does not include version information.
+            If the version number in the configuration file is not equal to 1.0.
+
+        Returns
+        -------
+        aconfig : configparser.ConfigParser
+            Configuration for the D-FAST Morphological Impact analysis with only absolute paths.
+        """
+        config = configparser.ConfigParser()
+        with Path(filename).open("r", encoding="utf-8") as configfile:
+            config.read_file(configfile)
+
+        file_version = config.get("General", "Version", fallback= "")
+        if len(file_version) == 0 :
+            raise LookupError("No version information in the file!")
+
+        if not (Version(file_version) == Version("1") or Version(file_version) == Version("2")):
```

Comment:
Maybe something we can thing about?
```suggestion
        if not (Version(file_version) == Version("1") or Version(file_version) == Version("2") or Version(file_version) == Version("3") or Version(file_version) == Version("4") or Version(file_version) == Version("5") or Version(file_version) == Version("etc..")):
```

Reply:
not in scope. see DFAST-108